source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -52,52 +56,34 @@ public PimsCompensationRequisition GetById(long compensationRequisitionId)
-            compensationRequisition.ThrowIfNull(nameof(compensationRequisition));
-
-
-            /* TODO: Fix compensation requisition
-            _user.ThrowInvalidAccessToAcquisitionFile(_userRepository, _acqFileRepository, compensationRequisition.AcquisitionFileId);
-
-            _logger.LogInformation($"Updating Compensation Requisition with id ${compensationRequisition.CompensationRequisitionId}");
-
-            var currentCompensation = _compensationRequisitionRepository.GetById(compensationRequisition.CompensationRequisitionId);
-
-            var currentAcquisitionFile = _acqFileRepository.GetById(currentCompensation.AcquisitionFileId);
-            var currentAcquisitionStatus = Enum.Parse<AcquisitionStatusTypes>(currentAcquisitionFile.AcquisitionFileStatusTypeCode);
-
-            if (!_statusSolver.CanEditOrDeleteCompensation(currentAcquisitionStatus, currentCompensation.IsDraft) && !_user.HasPermission(Permissions.SystemAdmin))
+            PimsCompensationRequisition newCompensationRequisition = fileType switch
             {
-                throw new BusinessRuleViolationException("The file you are editing is not active or hold, so you cannot save changes. Refresh your browser to see file state.");
-            }
-
-            CheckTotalAllowableCompensation(currentAcquisitionFile, compensationRequisition);
-            compensationRequisition.FinalizedDate = CheckFinalizedDate(currentCompensation.IsDraft, compensationRequisition.IsDraft, currentCompensation.FinalizedDate);
+                FileTypes.Acquisition => AddAcquisitionFileCompReq(compensationRequisition),
+                FileTypes.Lease => AddLeaseFileCompReq(compensationRequisition),
+                FileTypes.Disposition => throw new BadRequestException("Relationship type not valid."),
+                FileTypes.Research => throw new BadRequestException("Relationship type not valid."),
+                _ => throw new BadRequestException("Relationship type not valid."),
+            };
 
-            PimsCompensationRequisition updatedEntity = _compensationRequisitionRepository.Update(compensationRequisition);
-            AddNoteIfStatusChanged(compensationRequisition.Internal_Id, compensationRequisition.AcquisitionFileId, currentCompensation.IsDraft, compensationRequisition.IsDraft);
-            _compensationRequisitionRepository.CommitTransaction();
+            return newCompensationRequisition;
+        }
 
-            return updatedEntity;
+        public PimsCompensationRequisition Update(PimsCompensationRequisition compensationRequisition)
+        {
+            _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionEdit);
+            compensationRequisition.ThrowIfNull(nameof(compensationRequisition));
 
-            DateOnly? CheckFinalizedDate(bool? currentStatusIsDraft, bool? newStatusIsDraft, DateOnly? currentValue)
+            if (compensationRequisition.AcquisitionFileId is not null)
```

Comment:
same question here, why are we not using fileTypes?

Reply:
Added